### ISSUE TITLE
IMA extenstion

### ISSUE THREAD

**Issue by giladna** [google/ExoPlayer#3669]
Hi,

Wanted to describe a scenario in IMA extention ads playback,
the VMAP I use contains preroll, several cue points as midrolls and postroll.
User watches preroll then seeks to the end  of the stream. In the current implementation what will happen is that the user will see all the ads that are configured in the vamp and that he did not watch, and only after that the postroll will be shown
I think this behaviour should be configurable in some way.

I beleive that displaying only the last relevant cuepoint to the position after seek is the user friendly behaviour.
I do not see reason why we will display all the ads in the cue points in the VMAP till the postroll.
I would preffer to show only the postroll. If user then seeks back he will get the next ad on its correct time offset when it comes.

10x
Gilad,

**Comment by andrewlewis** [comment]
The intended behavior is as follows: when seeking after the last midroll before a postroll, the user should see the last midroll, then content will play to the end, then they will see the postroll. I think that's the same as the desired behavior you're describing. If there are two midrolls then a postroll and the user seeks after the second midroll, the first midroll should not be shown. (But if an ad break contains multiple ads in an ad pod, all the ads in the ad pod are shown as part of playing the ad break.)

It sounds like that's not consistent with what you're seeing. Could you provide a link to the ad tag you're trying to play (here or via email to `dev.exoplayer@gmail.com`) and we can take a look? As we're wrapping the IMA SDK we have limited control over rules concerning what ads to play, but if the behavior is not as described this sounds like it might be an IMA bug. Thanks!

**Comment by andrewlewis** [comment]
The ExoPlayer demo app contains a sample ad tag "VMAP pre-roll single ad, mid-roll standard pods with 5 ads every 10 seconds for 1:40, post-roll single ad" that works correctly for me: after the preroll plays, seek after the final midroll and you should see that the content plays to the end from the seek position, then the postroll plays.

**Comment by giladna** [comment]
@andrewlewis 
Thanks,
What I see when seek to end 
I have 2 ads in each cuepoint when I seek to end  (using nexus 6p 8.1.0)
the ads indexs are mixed and I see both the last cuepoint and the post roll
I think that the correct flow is to play only the post roll.

sending you now the adtag.

Another issue in this adtag that it is dynamic

in pre roll there are alwayes 2 ads you might get 2 short ads or one long with skip and one short

if you get the long one and skip it will not play the second ad in the pod (bumper) if you wait till the the end it will play it.

Thanks gilad

**Comment by giladna** [comment]
@andrewlewis 
Did you get the mail?
10x
Gilad.

**Comment by andrewlewis** [comment]
@giladna Yes, thanks. I'll take a look soon.

**Comment by andrewlewis** [comment]
Is the content longer than the last midroll position (~ 45 min)? I can't verify as the link from the email returns HTTP status 404. If not, perhaps that could explain the weird behavior. Otherwise, please provided an updated/working link to the content.

I'm not sure exactly how to handle the content being shorter than the last midroll position, but it seems preferable to prepare the ad tag such that all ad breaks lie within the content if possible.

**Comment by giladna** [comment]
@andrewlewis 
Thanks,
It might be possible that the m3u8 has geo blocking.
The video durationI have tested is 60 min so it is not that case.

I am sending another m3u8 url to the ExoPlayer dev mail so you can test it with working stream.

BR,
Gilad.

**Comment by giladna** [comment]
@andrewlewis 
did you check it again?
10x

**Comment by andrewlewis** [comment]
@giladna I'm afraid not, but I haven't forgotten this and will try to get round to it soon.

**Comment by andrewlewis** [comment]
I tried to reproduce this using the ad tag and content provided via email but the ad groups seem to be empty and I only see a preroll and two midroll ad groups (no postroll). Please could you check the ad tag? Thanks.
